core/src/test/java/org/infinispan/scattered/statetransfer/CoordinatorStopTest.java

```diff
@@ -0,0 +1,168 @@
+      rpcManager2.excludeCommands(StateResponseCommand.class);
+
+      // segment 0 will be moved to cache(2). Since we've lost coord cache(1) now -> 0 and cache(2) -> 1
+      chf.setOwnerIndexes(new int[][] { {1}, {0}, {1} });
+
+      log.infof("Stopping coordinator %s, last stable topology is %d", manager(0), stableTopologyId);
+      manager(0).stop();
+
+      log.infof("Current members are: %s", cache(1).getCacheManager().getMembers());
+      assertTrue(cache(1).getCacheManager().isCoordinator());
+
+      // topology + 1 is the one that just omits the leaving node
+      BlockedTopology t1 = ltm2.expectTopologyUpdate(CacheTopology.Phase.NO_REBALANCE, stableTopologyId + 1);
+      assertEquals(2, t1.getCacheTopology().getActualMembers().size());
+      assertEquals(null, t1.getCacheTopology().getPendingCH());
+      assertOwners(t1, true, 0);
+      assertOwners(t1, true, 1, address(1));
+      assertOwners(t1, true, 2, address(2));
+      t1.unblock();
+
+      // topology + 2 has TRANSITORY phase and all segments have an owner in pendingCH
+      BlockedTopology t2 = ltm2.expectTopologyUpdate(CacheTopology.Phase.TRANSITORY, stableTopologyId + 2);
+      assertEquals(2, t2.getCacheTopology().getActualMembers().size());
+      assertNotNull(t2.getCacheTopology().getPendingCH());
+      assertOwners(t2, false, 0, address(2));
+      assertOwners(t2, false, 1, address(1));
+      assertOwners(t2, false, 2, address(2));
+      t2.unblock();
+
+      // Let the rebalance begin
+      rpcManager2.expectCommand(StateRequestCommand.class,
+            request -> assertEquals(StateRequestCommand.Type.CONFIRM_REVOKED_SEGMENTS, request.getType()))
+            .send().receiveAll();
+      // Allow cache(2) to receive the view
+      viewLatch.countDown();
+
+      ControlledRpcManager.BlockedRequest keyTransferRequest = rpcManager2.expectCommand(StateRequestCommand.class,
+            request -> assertEquals(StateRequestCommand.Type.START_KEYS_TRANSFER, request.getType()));
+
+      // topology + 3 should have null pendingCH
+      BlockedTopology t3 = ltm2.expectTopologyUpdate(CacheTopology.Phase.NO_REBALANCE, stableTopologyId + 3);
+      assertEquals(2, t3.getCacheTopology().getActualMembers().size());
+      assertEquals(null, t3.getCacheTopology().getPendingCH());
+
+      CountDownLatch topologyLatch = new CountDownLatch(1);
+      cache(2).getAdvancedCache().addListener(new TopologyChangeListener(topologyLatch));
+      t3.unblock();
+
+      ControlledRpcManager.BlockedRequest cancelStateTransfer = rpcManager2.expectCommand(StateRequestCommand.class,
+            request -> assertEquals(StateRequestCommand.Type.CANCEL_STATE_TRANSFER, request.getType()));
+      cancelStateTransfer.sendWithoutResponses();
+
+      // Wait until topology + 3 is installed
+      assertTrue(topologyLatch.await(10, TimeUnit.SECONDS));
+
+      // unblock outdated rebalance
+      rpcManager2.stopBlocking();
+      keyTransferRequest.send().receiveAll();
```

Comment:
This is the kind of thing I meant in the chat... the fact that you call `stopBlocking()` before completing the blocked request makes me think that the test absolutely needs another command to be invoked remotely, but you'd rather not say what command :)

Reply:
I need to verify if this works (I didn't ran the test manually after last update and there was an oversight I had to fix); I've used similar construct in other test and seems it got broken as receiveAll schedules the continuation to the rpcManager's executor.

Reply:
I think I understand now. In my mind, the only alternative would have been to stop blocking at the end of the test, and stopping the executor wouldn't matter there because all the commands are already done.